common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -1167,5 +1168,5 @@ def _update_module_location(module, new_location):
     # explicitly set each field to its current value before triggering the save.
     if len(rekey_fields) > 0:
         for rekey_field_name in rekey_fields:
-            setattr(module, rekey_field_name, getattr(module, rekey_field_name))
+            module.fields[rekey_field_name].force_dirty(module)
```

Comment:
Also, looking at this code (and the comment above), it seems like maybe the right thing would be for XBlock to notice when `scope_ids` change, so that it will automatically re-submit all of the fields.